sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -27,7 +27,10 @@ import com.squareup.kotlinpoet.CodeBlock
 import com.squareup.kotlinpoet.FunSpec
 import com.squareup.kotlinpoet.NameAllocator
 
-abstract class QueryGenerator(private val query: BindableQuery) {
+abstract class QueryGenerator(
+  private val query: BindableQuery,
+  private val treatNullAsUnknownForEquality: Boolean
+) {
   protected val dialect = query.statement.sqFile().dialect
```

Comment:
Instead of passing the flag everywhere we should get it through the project service similar to what we do for the dialect.

Reply:
if you want to make it possible to to override in tests you can leave as a constructor param with a default value. Mostly just dont want to see it getting passed around everywhere, it will clean up the diff a lot

Reply:
So add it to `SqlDelightFile` in the same was that `dialect` is?

And is that specifically in the context of `QueryGenerator`, `SelectQueryGenerator`, `MutatorQueryGenerator`, and `ExecuteQueryGenerator`. or are there other places it should go through the project service?

Reply:
specifically `QueryGenerator`, but then once you've changed that you can update all the callsites to not pass it in and that should propagate throughout everything.

Yep, in `SqlDelightFile`